shell/etc/fish/conf.d/conda.fish

```diff
@@ -217,14 +217,17 @@ end
 #   command conda env list | awk 'NR > 2 {print $1}'
 # end
 
+set -x CONDA_ROOT_FISH_CACHED (command conda info --root)
 
 # Faster but less tested (?)
 function __fish_conda_commands
-  command echo -e "activate\ndeactivate" ;and ls --color=none (conda info --root)/bin/conda-* | sed -r 's/^.*conda-([a-z]+)/\1/'
+  string replace -r '.*_([a-z]+)\.py$' '$1' $CONDA_ROOT_FISH_CACHED/lib/python*/site-packages/conda/cli/main_*.py
+  echo activate
+  echo deactivate
 end
 
 function __fish_conda_envs
-  command echo root ;and ls -1 --color=none (conda info --root)/envs/
+  basename  $CONDA_ROOT_FISH_CACHED/envs/*
```

Comment:
My `basename` (basename (GNU coreutils) 8.28) needs `-a` for multiple entries. Also, should append a trailing slash to avoid listing files, like this:

```
function __fish_conda_envs
  basename -a $CONDA_ROOT_FISH_CACHED/envs/*/
end
```